src/sagemaker/tensorflow/estimator.py

```diff
@@ -245,7 +289,11 @@ def fit_super():
         if run_tensorboard_locally and wait is False:
             raise ValueError("Tensorboard is not supported with async fit")
 
-        if run_tensorboard_locally:
+        if self._script_mode_enabled():
+            if run_tensorboard_locally:
+                LOGGER.warning(_SCRIPT_MODE_TENSORBOARD_WARNING.format(self.model_dir))
+            fit_super()
+        elif run_tensorboard_locally:
```

Comment:
Maybe this is simpler?

```python 
        if run_tensorboard_locally:
            tensorboard = Tensorboard(self)
            tensorboard.validate_requirements()
              try:
                 tensorboard.start()
                 fit_super()
             finally:
                 # sleep 20 secs for tensorboard start up if fit() quits instantly
                 time.sleep(20)
                 tensorboard.event.set()
                 tensorboard.join()
         else:
            if self._script_mode_enabled():
                LOGGER.warning(_SCRIPT_MODE_TENSORBOARD_WARNING.format(self.model_dir))
             fit_super()
```

or even

```python 
              try:
                  if run_tensorboard_locally:
                       tensorboard = Tensorboard(self)
                       tensorboard.validate_requirements()
                       tensorboard.start()
             finally:
                  if run_tensorboard_locally:
                       # sleep 20 secs for tensorboard start up if fit() quits instantly
                       time.sleep(20)
                       tensorboard.event.set()
                       tensorboard.join()
  
                  if self._script_mode_enabled():
                       LOGGER.warning(_SCRIPT_MODE_TENSORBOARD_WARNING.format(self.model_dir))
                  fit_super()
```

Reply:
These two doesn't read much better to me. I have combined the first two ifs. Let me know if you feel strongly about this. :)